websites/C/Coub/presence.ts

```diff
@@ -1,247 +1,370 @@
-        data.state = "Top coubs";
-      } else if (Routes[2] === "views") {
-        data.state = "Views count coubs";
-      } else if (Routes[2] === "random") {
-        data.state = "Random coubs";
+          if (showBridge)
+            data.buttons.push({
+              label: "Download via Bridge",
+              url: parseBridgeUrl(activeMedia.dataset.permalink)
+            });
+          return data;
+        }
+      },
+      {
+        middleware: (ref) => !!ref.window,
+        exec: (
+          context,
+          data,
+          { strings }: { strings: { browsing: string } }
+        ) => {
+          if (!context) return null;
+          data.state = strings.browsing;
+          data.details = "";
+          return data;
+        }
       }
-    }
-  } else {
-    data.details =
-      "Channel: " +
-      document.querySelector(".channel .channel__description h1").textContent;
+    ],
+    presence = new Presence({
+      clientId: "818598086984728576"
+    });
 
-    if (Routes[1] === "edit") {
-      data.state = "Edit channel";
-    } else {
-      data.state = document.querySelector(
-        ".profile__menu .page-menu__item.-active"
-      ).textContent;
-    }
-  }
+  (function (app: Presence) {
+    app.on("UpdateData", async () => {
+      const presenceData: PresenceData = {
+        largeImageKey: "logo",
+        largeImageText: "Coub"
+      } as PresenceData;
+      if (document.location.hostname == "coub.com") {
+        const query: { [key: string]: unknown } = getQuery(),
+          strings: { [key: string]: string } = await app.getStrings({
+            play: "presence.playback.playing",
+            pause: "presence.playback.paused",
+            browsing: "presence.activity.browsing",
+            searching: "presence.activity.searching",
+            watching: "presence.playback.playing"
+          }),
+          context = pages.find((x) => x.middleware(window, [query]));
+        if (!context) return false;
 
-  presence.setActivity(data, true);
-});
+        const result = Promise.resolve(
+          context.exec(app, presenceData, {
+            strings,
+            query,
+            showWatch: await app
+              .getSetting("show_button_watching")
+              .then((x) => !!x)
+              .catch(() => true),
+            showBridge: await app
+              .getSetting("show_coubbridge")
+              .then((x) => !!x)
+              .catch(() => false)
+          })
+        );
+        return result.then((data) => {
+          if (!data) {
+            presence.setTrayTitle();
+            presence.setActivity({
+              largeImageKey: "logo",
+              state: strings.browsing
+            });
+          } else {
+            if (data.details) presence.setActivity(data);
+            if (data.buttons && data.buttons.length === 0) delete data.buttons;
+          }
+          return data;
+        });
+      }
+    });
+  })(presence);
+})();
```

Comment:
just use the normal updatedata event.,..?

Reply:
im using it? just using better structure for several pages.